<a href="https://colab.research.google.com/github/sangeun0/Learning-python/blob/main/4_2_practice_loop2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 문제 1.
## 1부터 100까지의 정수 중 한 개를 입력받아 100 보다 작은 배수들을 차례로 출력하다가 10 의 배수가 출력되면 프로그램을 종료하도록 프로그램을 작성하시오.
- 입력 예: 7
- 출력 예: 7 14 21 28 35 42 49 56 63 70

In [ ]:
#숫자입력
number1= int(input( "enter the number: "))
i=0

#몫의 범위 지정
for i in range(1,9):
  i=i+1
  mul= number1*i
  if int(mul)%10 ==number1: #배수값을 10으로 나누었을 때 입력받은 값이 나오면 프로그램 종료
    break
  print(mul)

enter the number: 7
14
21
28
35
42
49
56
63


IndentationError: ignored

enter the number:6
6
12
18
24
30
36


입력: 13
출력: 13 26 39 52 65 78 91 

# 문제 2.
## 자연수 n을 입력받아 "출력 예"와 같이 공백으로 구분하여 출력하는 프로그램을 작성하시오.
- 입력 예: 3

In [ ]:
    1
  1 2
1 2 3

    1
  1 3
1 3 5
    

IndentationError: ignored

In [ ]:
n= int(input("enter the number: "))
# 숫자 입력
for i in range(1,n+1):
  #줄수 =n개
  for j in range(1,n+1): #열수 =n개
    if j> n+1-(i+1): # 자리값이 커야 숫자가 프린트 되도록 설정
      print(j-((n+1)-(i+1)), end="") #숫자가 n값에 가가울 수록 커지게 +1씩 되면서 출력되도록
    else:
      print(" ", end="") #자리값이 작으면 공백으로 채우도록 설정
  print()

enter the number: 3
  1
 12
123


1
2
3
4
5



# 문제 3.
## 자연수 n을 입력받아 "출력 예"와 같이 공백으로 구분하여 출력되는 프로그램을 작성하시오.
- 입력 예 : 3

In [ ]:
1 2 3
  4 5
    6
    
1 3 5
  7 9
    11    

SyntaxError: ignored

In [101]:
n= int(input("enter the number: "))
# 숫자 입력
for i in range(n+1, 1 ,-1):
  #줄수 =n개
  for j in range(1,n+1): #열수 =n개
    if j> n+1-(i+1): # 자리값이 커야 숫자가 프린트 되도록 설정
      print(, end="") #숫자가 n값에 가가울 수록 커지게 +1씩 되면서 출력되도록
    else:
      print(" ", end="") #자리값이 작으면 공백으로 채우도록 설정
  print()

enter the number: 3
234
123
 12


enter the number: 3
2 


4 3 2 


# 문제 4.
## 10개의 정수를 입력받아 100 미만의 수 중 가장 큰 수와 100 이상의 수 중 가장 작은 수를 출력하는 프로그램을 작성하시오.
- List 데이터 구조를 활용하시오.
- 입력 예: 88 123 659 15 443 1 99 313 105 48
- 출력 예: 99 105

In [ ]:
list1 = [];
list2 = [];
#리스트 설정
for i in range(10):
    num = input("enter the number") #입력값
    if int(num) >= 100:
        list1.append(num) #100이상 수 list1에 넣기
    else: #나머지 list2에 넣기
        list2.append(num)

maxNum = max(list2) #list2 중 최대값
minNum = min(list1) #list1 중 최대값
print(maxNum, minNum)

enter the number32
enter the number435
enter the number12
enter the number101
enter the number354
enter the number23
enter the number21
enter the number76
enter the number5
enter the number2
76 101


enter the number32
enter the number43
enter the number101
enter the number123
enter the number45
enter the number999
enter the number234
enter the number56
enter the number34
enter the number1
56 101


# 문제 5.
## 리스트을 선언한 후 정수를 차례로 입력 받다가 -1이 입력되면 입력을 중단하고, -1을 제외한 마지막 세 개의 정수를 출력하는 프로그램을 작성하시오. 
- 입력받은 정수가 -1을 제외하고 3개 미만일 경우에는 -1을 제외하고 입력받은 정수를 모두 출력한다.
- 입력 예: 30 20 10 60 80 -1
- 출력 예: 10 60 80

In [ ]:
number=[]

while:
  number= input("enter the number: ")
  if number == -1:
    break
  if number>0:
    print()

입력: 10
입력: 20
입력: 30
입력: 60
입력: 20
입력: -1
[30, 60, 20]


# 문제 6. 
## 사용자에게 base문자열을 첫 줄에 입력받고,  다음으로 찾고자하는 문자 1개를 입력받아 문자열 내의 검색 문자의 모든 위치를 출력하는 프로그램을 작성하시오.
- 첫 번째 위치는 0번이며 리스트에 없는 문자가 입력되면 "none" 라는 메시지를 출력하고 끝내는 프로그램을 작성하시오.
- 예제: "my name is dean" 을 base 문자열로 입력하고, 찾고자 하는 문자를 'd'로 입력하면 11을 출력해주면 된다.
- 만약, 찾으려는 문자가 2개 이상인 경우에는 모든 위치를 출력해준다.

In [91]:
str = 'abcabcabc'
target = 'b'
index = -1
while True:
    index = str.find(target, index + 1)
    if index == -1:
        break
    print('start=%d' % index)
    



start=1
start=4
start=7


In [95]:
base= input("enter the string: ")
c= input("enter the character: ")
index = -1
while True:
    index = base.find(c, index+1) #원하는 문자가 몇 번째에 있는지 =find함수
                                    #find(찾을 요소, 시작위치)
    if index == -1: #지정한 문자를 찾지 못했을 경우
      break
    print(index)

enter the string: name apple
enter the character: a
1
5


# 문제 7.
## 학생들의 점수를 입력을 받다가 0이 입력되면 그 때까지 입력받은 점수를 10점 단위로 구분하여 점수대별 학생 수를 출력하는 프로그램을 작성하시오. 
- 1명도 없는 점수는 출력하지 않는다.
- 입력 예: 63 80 95 100 46 64 88 0

In [ ]:
출력 예: 
100 : 1 person
90 : 1 person
80 : 2 person
60 : 2 person
40 : 1 person

In [ ]:
count=0
count1=0
count2=0
count3=0
count4=0
score=[]
#갯수 값 전부 0으로 초기화
while True:
  x= input("enter the score: ")
  score.append(x)
#리스트 값에는 int사용할 수 없어 if문 마다 x값에 int 
  if int(x) == 0:
    break
  elif int(x)== 100:
    count= count+ 1
  elif int(x)>=90 and int(x)<100:
    count1=count1+1
  elif int(x)>=80 and int(x)<90:
    count2=count2+1
  elif int(x)>=60 and int(x)<80:
    count3=count3+1
  elif int(x)<=40 and int(x)<60:
    count4=count4+1


  print("100", count ,":person")
  print("90", count1 ,":person")
  print("80", count2, ":persin")
  print("60", count3 ,":person")
  print("40", count4 ,":person")

enter the score: 0


# 문제 8.
## 소수(prime number)란 1보다 큰 자연수 중 1과 자기 자신 두 개만을 약수로 갖는 수를 말한다. 
- 자연수 M과 N을 입력받아 M부터 N까지 소수의 개수를 구하여 출력하는 프로그램을 작성하시오.
- M이상 N이하의 자연수 중 소수가 몇 개인지 구하여 출력한다.
- 입력 예: 10 100
- 출력 예: 21

In [ ]:
m= int(input("enter the number1"))
n= int(input("enter the number2"))
count=0
for i in range(m,n+1):
  prime= True #범위내의 모든 숫자가 소수라고 가정
  for k in range(2,i):
    if i%k ==0: #자기 자신 전까지의 숫자로 나누었을때 나누어 떨어지면 소수가 아님
      prime= False
      continue
  if prime :
    count+=1 #남은 경우는 모두 소수
print(count)

enter the number110
enter the number2100
21


# 보너스
## 자연수 M과 N이 주어질 때 M이상 N이하의 자연수 중 소수인 것을 모두 골라 이들 소수의 합과 최소값을 찾는 프로그램을 작성하시오.

- 예를 들어 M=60, N=100이 경우 60이상 100이하의 자연수 중 소수는 61, 67, 71, 73, 79, 83, 89, 97 총 8개가 있으므로, 
- 이들 소수의 합은 620이고, 최소값은 61이 된다.

# 문제 9.
## 아래의 결과를 보이도록 프로그램을 작성하시오.

In [ ]:
   *
  ***
 *****
*******

In [ ]:

for i in range(0, 5):
     print (" "* (5-i), "*" * (2*i-1)) #공백은 1개씩 작아짐, *은 두개씩 커짐

      
     *
    ***
   *****
  *******


# 문제 10.
## 아래의 피라미드 모형을 출력하시오.
- 출력 결과에서 #은 공백(space)로 처리해도 됨.

In [ ]:
#위쪽 피라미드
for i in range(1, 4+1):
    print (" "* (4-i), "*" * (2*i-1))  #공백-1, 별+2
#아래쪽 피라미드
for j in range(4-1, 0, -1) :
    print (" "* (4-j), "*" * (2*j-1)) #공백 +1, 별-2

    *
   ***
  *****
 *******
  *****
   ***
    *


In [ ]:
#위쪽 피라미드
for i in range(1, 2+1):
    print (" "* (2-i), "*" * (2*i-1))  #공백-1, 별+2
#아래쪽 피라미드
for j in range(2-1, 0, -1) :
    print (" "* (2-j), "*" * (2*j-1)) #공백 +1, 별-2

  *
 ***
  *


# 보너스2
## 삼각형의 높이 N을 입력받아서 아래와 같이 문자 'A'부터 차례대로 왼쪽 대각선으로 채워서 삼각형 모양을 출력하는 프로그램을 작성하시오.
- 입력 예 : 5

In [ ]:
        A
      B F
    C G J
  D H K M
E I L N O

# 문제 11.
## 2진수를 입력받아 10진수로 변환하여 출력하는 프로그램을 작성하시오.
- 입력 예: 10101
- 출력 예: 21

In [ ]:
number= input("enter the number: ")
number= number[::-1]
k = 0

# print(number)
# print(len(number))

for i in range(len(number)):
    if int(number[i]) == 1 :
        k += 2**i

print(k)

enter the number: 11
3


# 문제 12.

## 복호화 키와 암호 문자를 입력으로 받아 원문을 구하는 프로그램을 구현하시오. 복호화 키는 26개의 소문자로 주어지며, a,b,c,d... 를 순서대로 복호화 키 문자로 대치한다는 것을 뜻한다.

 

- 예를 들어, 복호화 키가 "eydbkmiqugjxlvtzpnwohracsf" 와 같이 주어진다고 하자, 

- 그러면 이는 다음과 같다 - a 문자는 e, b 문자는 y, ..., z 문자는 f로 바꿔줌.

- 암호화 된 문자는 대소문자 혹은 공백이 올수 있고 대소문자는 대문자로 소문자는 소문자로 치환 규칙에 맞게 출력하고 , 

- 공백문자는 그대로 출력한다.

- 입력 예: 
- eydbkmiqugjxlvtzpnwohracsf
- Kifq oua zarxa suar bti yaagrj fa xtfgrj
- 출력 예:
- Jump the fence when you seeing me coming

# diamond code

Enter a floor of diamond(odd)[quit: -1]: 3
  *  
 *** 
  *  
Enter a floor of diamond(odd)[quit: -1]: 1
*
Enter a floor of diamond(odd)[quit: -1]: 5
    *    
   ***   
  *****  
   ***   
    *    
Enter a floor of diamond(odd)[quit: -1]: 7
      *      
     ***     
    *****    
   *******   
    *****    
     ***     
      *      
Enter a floor of diamond(odd)[quit: -1]: -1


In [ ]:

n= int(input("enter the number: "))

for x in range(0, n):
  for y in range(0,n-2):
    print(" "*(1-i), "*"*(2*i+1),end="")

enter the number: 3
 ******* ******* *******

In [ ]:
x= int(input("enter the number:"))

for i in range(0, x):
     print (" "* (x-i), "*" * (2*i-1)) #공백은 1개씩 작아짐, *은 두개씩 커짐

enter the number:5
      
     *
    ***
   *****
  *******


mid: 0
*
mid: 1
 * 
***
 * 
mid: 2
  *  
 *** 
*****
 *** 
  *  
mid: 3
   *   
  ***  
 ***** 
*******
 ***** 
  ***  
   *   
mid: 4
    *    
   ***   
  *****  
 ******* 
*********
 ******* 
  *****  
   ***   
    *    


In [ ]:
#위쪽 피라미드
n= int(input("enter the number: "))

for i in range(0, n+1):
    print (" "* (n-i), "*" * (2*i+1))  #공백-1, 별+2
#아래쪽 피라미드
for j in range(n, 0, -1) :
    print (" "* ((n-j)+1), "*" * (2*j-1)) #공백 +1, 별-2

enter the number: 0
 *
